src/components/GameView/ChooseWhetherToCounterDialog.vue

```diff
@@ -0,0 +1,118 @@
+<template>
+  <base-dialog v-model="show" id="counter-dialog" title="Chance to Counter">
+    <template #body>
+      <div v-if="!opponentLastTwo" class="my-2">
+        Your opponent has played the 
+        <game-card-name :card-name="oneOff.name" />
+        as a one-off
+        <span v-if="target"> 
+          targeting your 
+          <game-card-name :card-name="target.name" />
+        </span>
+      </div>
+      <div v-else class="my-2">
+        Your opponent has played 
+        <game-card-name :card-name="opponentLastTwo.name" />
+
+        to Counter
+        <span v-if="playerLastTwo">
+          your 
+          <game-card-name :card-name="playerLastTwo.name" />
+        </span>
+      </div>
+      <div class="d-flex justify-center align-center my-8">
+        <game-card :suit="oneOff.suit" :rank="oneOff.rank" />
+        <p class="ml-8">
+          {{ oneOff.ruleText }}
+        </p>
+        <div v-if="target" id="target-wrapper">
+          <span id="target-icon-wrapper" class="d-flex justify-center align-center">
+            <v-icon size="x-large" color="red" icon="mdi-target" />
+          </span>
+          <game-card :suit="target.suit" :rank="target.rank" />
+        </div>
+      </div>
+      Would you like to play a two to counter?
+    </template>
+
+    <template #actions>
+      <v-btn data-cy="decline-counter-resolve" color="surface-1" variant="outlined" @click="resolve" class="mr-4">
+        Resolve
+      </v-btn>
+      <v-btn data-cy="counter" color="surface-1" variant="flat" @click="$emit('choose-to-counter')">
+        Counter
+      </v-btn>
+    </template>
+  </base-dialog>
+</template>
+
+<script>
+import BaseDialog from '@/components/Global/BaseDialog.vue';
+import GameCard from '@/components/GameView/GameCard.vue';
+import GameCardName from '@/components/GameView/GameCardName.vue';
+
+export default {
+  name: 'ChooseWhetherToCounterDialog',
+  components: {
+    BaseDialog,
+    GameCard,
+    GameCardName,
+  },
+  emits: ['choose-to-counter', 'resolve'],
+  props: {
+    modelValue: {
+      type: Boolean,
+      required: true,
+    },
+    oneOff: {
+      type: Object,
+      required: true,
+    },
+    target: {
+      type: Object,
+      default: null,
+    },
+    twosPlayed: {
+      type: Array,
+      required: true,
+    },
+  },
+  computed: {
+    show: {
+      get() {
+        return this.modelValue;
+      },
+      set() {
+        // do nothing - parent controls whether dialog is open
+      },
+    },
+    opponentLastTwo() {
+      return this.twosPlayed && this.twosPlayed.length > 0
+        ? this.twosPlayed[this.twosPlayed.length - 1]
+        : null;
+    },
+    playerLastTwo() {
+      return this.twosPlayed && this.twosPlayed.length > 1
+        ? this.twosPlayed[this.twosPlayed.length - 2]
+        : null;
+    },
```

Comment:
There's a level of duplication between this and the other component that smells. It's fine for now but longterm this feels like a problem waiting to happen.

Reply:
Since we have the HoC I think the move is to compute these in the parent and send them as a prop to keep the logic centralized

Reply:
Actually that only applies if the computed props are used in both children. I'm silly both of these are only need in one of them. I'm nixing this from the other too. Silly me 😅